ceph/iscsi/initiator.py

```diff
@@ -0,0 +1,122 @@
+from ceph.ceph_admin.common import config_dict_to_string
+
+
+class ISCSIConfigs:
+    """iSCSI Configuration attributes."""
+
+    INITIATOR_IQN = "InitiatorName={IQN}"
+    ISCSI_IQN_PATH = "/etc/iscsi/initiatorname.iscsi"
+    MPATH_CONFIG_FILE_PATH = "/etc/multipath.conf"
+    ISCSID_CONFIG_FILE_PATH = "/etc/iscsi/iscsid.conf"
+    ISCSI_PACKAGES = "iscsi-initiator-utils device-mapper-multipath"
+    MPATH_CONFIG = """
+devices {
+    device {
+            vendor                 "LIO-ORG"
+            product                "TCMU device"
+            hardware_handler       "1 alua"
+            path_grouping_policy   "failover"
+            path_selector          "queue-length 0"
+            failback               60
+            path_checker           tur
+            prio                   alua
+            prio_args              exclusive_pref_bit
+            fast_io_fail_tmo       25
+            no_path_retry          queue
+    }
+}
+"""
+
+
+class ISCSIInitiator(ISCSIConfigs):
+
+    def __init__(self, node, iqn):
+        """iSCSI Initiator module.
+
+        Args:
+            node: Ceph Node object
+            iqn: Initiator's iSCSI Qualified Name
+        """
+        self.node = node
+        self._iqn = iqn
+        self.iscsiadm = self.ISCSIadmin(self)
+        self.multipath = self.Multipath(self)
+        self.configure()
+
+    @property
+    def iqn(self):
+        return self._iqn
+
+    @iqn.setter
+    def iqn(self, value):
+        self._iqn = value
+
+    def configure(self, multipath=True):
+        """Configure iSCSI initiator."""
+        # Install iscsi utils
+        configure_cmds = [f"yum install -y fio {self.ISCSI_PACKAGES}"]
```

Comment:
`fio`??? I hope it does not defeat the purpose of the class.

Likewise should we use `dnf` instead of `yum` though both works.

Reply:
Yeah I could take fio out of the class.

Reply:
Done.